frontend/src/components/Contexts/MissionListsContext.tsx

```diff
@@ -31,9 +39,77 @@ const fetchMissions = (params: {
 }): Promise<Mission[]> => {
     return BackendAPICaller.getMissionRuns(params).then((response) => response.content)
 }
-export const useMissions = (refreshInterval: number): MissionsResult => {
+export const useMissions = (): MissionsResult => {
     const [ongoingMissions, setOngoingMissions] = useState<Mission[]>([])
     const [missionQueue, setMissionQueue] = useState<Mission[]>([])
+    const { registerEvent, connectionReady } = useSignalRContext()
+
+    useEffect(() => {
+        if (connectionReady) {
+            registerEvent('mission run created', (username: string, message: string) => {
+                const newMission: Mission = JSON.parse(message)
+                if (missionQueue.find((m) => m.id === newMission.id))
+                    setMissionQueue((oldQueue) => [...oldQueue, newMission])
+                else
+                    setMissionQueue((oldQueue) => {
+                        let missionQueueCopy = [...oldQueue]
+                        missionQueueCopy = upsertList(missionQueueCopy, newMission)
+                        return [...missionQueueCopy]
+                    })
```

Comment:
Is this if statement correct? By the first check it looks like you're checking if the new mission already exists and then instead of upserting you are adding a new version of it at the end of the list?

Reply:
I've fixed it now, good catch